ui/apps/platform/src/utils/searchUtils.ts

```diff
@@ -107,6 +107,16 @@ export function searchOptionsToSearchFilter(searchOptions: SearchEntry[]): Searc
     return searchFilter;
 }
 
+/**
+ * Determines whether or not a SearchFilter contains a valid value for
+ * all keys. A valid value is either a non-empty string or non-empty array.
+ */
+export function isCompleteSearchFilter(searchFilter: SearchFilter) {
+    return Object.values(searchFilter).every(
+        (o) => Boolean(o) && (!Array.isArray(o) || o.length > 0)
+    );
+}
```

Comment:
I think there is similar functionality in the WorkflowState class. 
(Unfortunately, that is written in classic OO style, and has become something of a "god-class", so I'm not sure it's worth trying to reconcile what's in it with helper functions like this one.)

Reply:
Yeah, maybe whenever attention turns back to workflow, we might factor out some workflow methods to become available to call as pure functions.